src/Aspire.Hosting.Azure.AppService/AzureAppServiceWebsiteContext.cs

```diff
@@ -0,0 +1,379 @@
+    {
+        ProcessEndpoints();
+
+        await ProcessEnvironmentAsync(cancellationToken).ConfigureAwait(true);
+        await ProcessArgumentsAsync(cancellationToken).ConfigureAwait(true);
+    }
+
+    private async Task ProcessEnvironmentAsync(CancellationToken cancellationToken)
+    {
+        if (resource.TryGetAnnotationsOfType<EnvironmentCallbackAnnotation>(out var environmentCallbacks))
+        {
+            var context = new EnvironmentCallbackContext(
+                environmentContext.ExecutionContext, resource, EnvironmentVariables, cancellationToken);
+
+            foreach (var c in environmentCallbacks)
+            {
+                await c.Callback(context).ConfigureAwait(true);
+            }
+        }
+    }
+
+    private async Task ProcessArgumentsAsync(CancellationToken cancellationToken)
+    {
+        if (resource.TryGetAnnotationsOfType<CommandLineArgsCallbackAnnotation>(out var commandLineArgsCallbackAnnotations))
+        {
+            var context = new CommandLineArgsCallbackContext(Args, cancellationToken)
+            {
+                ExecutionContext = environmentContext.ExecutionContext
+            };
+
+            foreach (var c in commandLineArgsCallbackAnnotations)
+            {
+                await c.Callback(context).ConfigureAwait(true);
+            }
+        }
+    }
+
+    private void ProcessEndpoints()
+    {
+        if (!resource.TryGetEndpoints(out var endpoints) || !endpoints.Any())
+        {
+            return;
+        }
+
+        // Only http/https are supported in App Service
+        var unsupportedEndpoints = endpoints.Where(e => e.UriScheme is not ("http" or "https")).ToArray();
+        if (unsupportedEndpoints.Length > 0)
+        {
+            throw new NotSupportedException($"The endpoint(s) {string.Join(", ", unsupportedEndpoints.Select(e => $"'{e.Name}'"))} on resource '{resource.Name}' specifies an unsupported scheme. Only http and https are supported in App Service.");
+        }
+
+        foreach (var endpoint in endpoints)
+        {
+            if (!endpoint.IsExternal)
+            {
+                throw new NotSupportedException($"The endpoint '{endpoint.Name}' on resource '{resource.Name}' is not external. App Service only supports external endpoints.");
+            }
+
+            // For App Service, we ignore port mappings since ports are handled by the platform
+            _endpointMapping[endpoint.Name] = new(
+                Scheme: endpoint.UriScheme,
+                Host: HostName,
+                Port: endpoint.UriScheme == "https" ? 443 : 80,
+                TargetPort: null, // App Service manages internal port mapping
+                IsHttpIngress: true,
+                External: true); // All App Service endpoints are external
+        }
+    }
+
+    private (object, SecretType) ProcessValue(object value, SecretType secretType = SecretType.None, object? parent = null)
```

Comment:
This logic can be largely reused between aca and app service. Endpoint resolution would need to be extracted somehow.